brave/brave-core
# Add support for Solana NFT in NFT detail screen

## Resolves https://github.com/brave/brave-browser/issues/27803 

⚠️ **Note:** change base branch from `simone/nft-detail-screen` to `master` as soon as it gets merged.

https://user-images.githubusercontent.com/3308503/213521595-d39cacd9-f035-4fae-80aa-66ad3e24fc5a.mov

## Details

Adds support for Solana NFT in NFT detail screen.
Compared to the ETH NFT the following changes have been introduced:
* Show `SPL` as new token standard
* Contract address label has become `Mint address`
* Mint address link points to NFT contract address using a different logic (consistent with iOS version)
* Mint address shown is truncated in the middle (consistent with iOS version)

### Light Theme

<img width="534" alt="Screenshot 2023-01-19 at 6 55 11 PM" src="https://user-images.githubusercontent.com/3308503/213522709-e3f7505a-982b-4c6c-b155-8fdde69d604f.png">

### Dark Theme

<img width="534" alt="Screenshot 2023-01-19 at 6 56 07 PM" src="https://user-images.githubusercontent.com/3308503/213522937-facfe90e-c1a1-407b-aebb-6c4b21789e9d.png">



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
- Open Brave wallet
- If not already selected, choose "Solana Mainnet Beta"
- Tap on "Edit visible assets" then "Add custom asset"
- Select `NFT` checkbox
- Add token address of a Solana NFT (e.g. `[C2b9dQ68jFRTX1xuMedq6FfKCjTWvdWnu8DPQC1zbmZq](https://opensea.io/assets/solana/C2b9dQ68jFRTX1xuMedq6FfKCjTWvdWnu8DPQC1zbmZq)`)
- Populate token name and token symbol
- Tap on "Add"
- Tap on "Save account watchlist"
- Tap on the newly added Solana NFT and open NFT detail screen
- Observe the image is loaded correctly (as of today, supported image types are GIF, BMP, PNG, JPG - Not SVG yet)
- Observe token address label is "Mint address" for Solana NFTs
- Observe token standard is "SPL" for Solana NFTs
- Observe the mint address is truncated in the middle.
- Tap on the mint address and observe the right NFT page is loaded correctly in the Solana blockchain explorer
